owasm/src/ext/crypto/alphavantage.rs

```diff
@@ -0,0 +1,86 @@
+//! [Alphavantage.co](https://www.alphavantage.co/) Oracle Extension
+
+use crate::core::{Oracle, ShellCmd};
+
+pub struct Price {
+    symbol: String,
+    alphavantage_api_key: String,
+}
+
+impl Price {
+    pub fn new(symbol: impl Into<String>, alphavantage_api_key: impl Into<String>) -> Price {
+        Price { symbol: symbol.into(), alphavantage_api_key: alphavantage_api_key.into() }
+    }
+}
+
+impl Oracle for Price {
+    type T = f32;
+
+    fn as_cmd(&self) -> ShellCmd {
+        ShellCmd::new(
+            "curl",
+            &[format!(
+                "https://www.alphavantage.co/query?function=GLOBAL_QUOTE&symbol={}&apikey={}",
+                &self.symbol, &self.alphavantage_api_key
+            )],
+        )
+    }
+
+    fn from_cmd_output(&self, output: String) -> Option<f32> {
+        let parsed = json::parse(&output).ok()?;
+        Some(parsed["Global Quote"]["05. price"].as_str()?.parse::<f32>().ok()?)
```

Comment:
Can we write as `parsed["Global Quote"]["05. price"].as_str()?.parse::<f32>().ok()`